lambda_src/drivers/destination_s3.py

```diff
@@ -31,9 +31,9 @@ def parse_destination_uri(destination: Text) -> Tuple[Text, Text]:
     LOG.debug(f'destination from header is {destination}')
     parsed_url = urlparse(destination)
     bucket = parsed_url.netloc
-    parts = parsed_url.path.split('/')
-    prefix = "/".join([x for x in parts if x]) 
-    prefix = time.strftime(prefix) if parsed_url.path.count('/') >= 1 else ''
+    parts = strftime(parsed_url.path).split('/')
+    prefix = "/".join([x for x in parts if x])
```

Comment:
what is the purpose of this `split` followed by a `filter` and a `join`? how would you explain this to users in the documentation? perhaps I'm missing something, but I would expect the logic here to be more straightforward, e.g. generate the file name to which to write with `{strftime(destination_uri)}-{batch_id}-{row}.json.gz`, unless `destination_uri` ends with a `/` in which case skip the first `-` and just put it in the folder specified by the user.